core/controllers/profile_test.py

```diff
@@ -1009,6 +1037,29 @@ def test_username_is_handled_correctly(self) -> None:
 
         self.logout()
 
+    def test_signup_with_user_has_already_registered(self) -> None:
```

Comment:
Your test function names should read like sentence clauses.

test_signup_with_user_who_has_already_registered

or

test_signup_with_already_registered_user

Also it's not really clear what you're testing here. Is it test_signup_with_already_registered_user_succeeds or test_signup_with_already_registered_user_fails or something else?

Please look at your function names throughout and apply the above to ensure that they are clear. Writing names that make sense is an important part of writing tests.